addons/parley/views/parley_graph_view.gd

```diff
@@ -351,4 +356,135 @@ func set_selected_by_id(id: String, _goto: bool = true) -> void:
 			set_selected(node)
 			_goto_node(node as ParleyGraphNode)
 			return
+
+#region SHORTCUTS
+
+
+const CLICK_DISTANCE: float = 50
+var onUnselect: Array[Callable] = []
+var selectedConnections: Array[ParleyGraphEdge] = []
+var selectedNodes: Array[ParleyGraphNode] = []
+
+func _on_node_selected(node: Node) -> void:
+	selectedNodes.append(node)
+func _on_node_deselected(node: Node) -> void:
+	selectedNodes.erase(node)
+	pass
+func _on_connections_deselected() -> void:
+	while onUnselect.size() > 0:
+		var callable: Callable = onUnselect.pop_front()
+		callable.call()
+
+func _gui_input(event: InputEvent) -> void:
+	if event is InputEventMouseButton:
+		_handle_mouse_select(event as InputEventMouseButton)
+
+	if event is InputEventKey:
+		var key_event: InputEventKey = event as InputEventKey
+		if key_event.pressed and not key_event.echo:
+			#Ctrl + Z
+			if key_event.ctrl_pressed and key_event.keycode == KEY_Z:
+				undo_request.emit()
+			#Ctrl + Y
+			elif key_event.ctrl_pressed and key_event.keycode == KEY_Y:
+				redo_request.emit()
+			#Ctrl + S
+			elif key_event.ctrl_pressed and key_event.keycode == KEY_S:
+				save_request.emit()
+			# Delete
+			elif key_event.keycode == KEY_DELETE:
+				delete_selected.emit()
+		
+func _handle_mouse_select(mouse_event: InputEventMouseButton) -> void:
```

Comment:
Just looking at this function logic, it seems a little quadratic. I wonder how this will behave for large dialogue sequences - especially since it's running on every click event, which is a very common operation. I suspect we may have performance issues with this, so I wonder if this is a feature we could maybe turn off in project settings and allow users to opt-in! wdyt?